tests/IceRpc.Tests/Transports/Slic/SlicTransportSslAuthenticationTests.cs

```diff
@@ -37,39 +38,20 @@ public async Task Ssl_client_connection_connect_fails_when_server_provides_untru
 
         // Start the TLS handshake.
         Task clientConnectTask = sut.Client.ConnectAsync(default);
-        Task? serverConnectTask = null;
-        IMultiplexedConnection? serverConnection = null;
-        try
-        {
-            // We accept two behaviors here:
-            // - the listener can internally kill the connection if it's not valid (e.g.: Quic behavior)
-            // - the listener can return the connection but ConnectAsync fails(e.g.: Slic behavior)
-            (serverConnection, _) = await listener.AcceptAsync(default);
-            serverConnectTask = serverConnection.ConnectAsync(default);
-        }
-        catch (AuthenticationException)
-        {
-            // Expected with Quic
-            serverConnectTask = null;
-        }
+        (IMultiplexedConnection serverConnection, _) = await listener.AcceptAsync(default);
+        var serverConnectTask = serverConnection.ConnectAsync(default);
 
         // Act/Assert
         Assert.That(async () => await clientConnectTask, Throws.TypeOf<AuthenticationException>());
 
-        // We accept two behaviors here:
-        // - serverConnectTask is null, the listener internally rejects the connection (e.g.: Quic behavior)
-        // - the server connect operation fails with an IceRpcException (e.g: Slic behavior).
-        if (serverConnection is not null)
-        {
-            // The client will typically close the transport connection after receiving AuthenticationException
-            await sut.Client.DisposeAsync();
-            var exception = Assert.ThrowsAsync<IceRpcException>(async () => await serverConnectTask!);
-            Assert.That(
-                exception?.IceRpcError,
-                Is.EqualTo(IceRpcError.ConnectionAborted).Or.EqualTo(IceRpcError.IceRpcError),
-                $"The test failed with an unexpected IceRpcError {exception}");
-            await serverConnection.DisposeAsync();
-        }
+        // The client will typically close the transport connection after receiving AuthenticationException
+        await sut.Client.DisposeAsync();
```

Comment:
I am confused here. 

provider will cleanup sut - we don't need these explicit DisposeAsync.

Do we need this `sut.Client.DisposeAsync` because without it the serverConnectTask won't fail?

Reply:
I guess we can remove the assert on the connection server side, otherwise, the server connect will not fail promptly

Reply:
I simplified the tests to only test what happens to the peer that is receiving the untrusted certificate, which is the point of this test. Testing what happens to the peer providing the untrusted certificate is not that interesting.